velox/exec/DistinctAggregations.cpp

```diff
@@ -185,6 +199,47 @@ class TypedDistinctAggregations : public DistinctAggregations {
     return input->template asUnchecked<RowVector>()->children();
   }
 
+  void extractForSpill(folly::Range<char**> groups, VectorPtr& result) const {
+    auto* arrayVector = result->as<ArrayVector>();
+    arrayVector->resize(groups.size());
+
+    auto* rawOffsets =
+        arrayVector->mutableOffsets(groups.size())->asMutable<vector_size_t>();
+    auto* rawSizes =
+        arrayVector->mutableSizes(groups.size())->asMutable<vector_size_t>();
+
+    size_t totalBytes = 0;
+    vector_size_t offset = 0;
+    for (auto i = 0; i < groups.size(); ++i) {
+      auto* accumulator =
+          reinterpret_cast<AccumulatorType*>(groups[i] + offset_);
+      rawSizes[i] = accumulator->maxSpillSize();
+      rawOffsets[i] = offset;
+      offset += accumulator->maxSpillSize();
+      totalBytes += accumulator->maxSpillSize();
+    }
+
+    auto& elementsVector = arrayVector->elements();
+    elementsVector->resize(totalBytes);
```

Comment:
It is super strange to create one element per byte... Am I missing something?

Reply:
I was trying to express a byte array really - an opaque set of bytes.

Reply:
I think in this case it should be ARRAY(VARBINARY) as it is an opaque set of bytes -per group-.  Let me submit a change addressing this resize call here to hopefully make it all clearer.